designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,940 @@
+
+```js
+const { ESLintRCCompat } = require("@eslint/eslintrc");
+
+const eslintrc = new ESLintRCCompat(__dirname);
+
+module.exports = [
+    "eslint:recommended",
+  
+    // load a file
+    eslintrc.extends("./.eslintrc.yml"),
+
+    // load eslint-config-standard
+    eslintrc.extends("standard"),
+
+    // load eslint-plugin-vue/recommended
+    eslintrc.extends("plugin:vue/recommended"),
+
+    // or multiple at once
+    eslintrc.extends("./.eslintrc.yml", "standard", "plugin:vue/recommended")
+
+];
+```
+
+#### Translating config objects
+
+The `ESLintRCCompat#config()` methods allows users to pass in a `.eslintrc`-style config and get back a config object that works with `eslint.config.js`. For example:
+
+```js
+const { ESLintRCCompat } = require("@eslint/eslintrc");
+
+const eslintrc = new ESLintRCCompat(__dirname);
+
+module.exports = [
+    "eslint:recommended",
+  
+    eslintrc.config({
+        env: {
+            node: true
+        },
+        root: true
+    });
+];
+```
+
+#### Including plugins
+
+The `ESLintRCCompat#plugins()` method allows users to automatically load a plugin's rules and processors without separately assigning a namespace. For example:
+
+```js
+const { ESLintRCCompat } = require("@eslint/eslintrc");
+
+const eslintrc = new ESLintRCCompat(__dirname);
+
+module.exports = [
+    "eslint:recommended",
+
+    // add in eslint-plugin-vue and eslint-plugin-example
+    eslintrc.plugins("vue", "example")
+];
+```
+
+This example includes both `eslint-plugin-vue` and `eslint-plugin-example` so that all of the rules are available with the correct namespace and processors are automatically hooked up to the correct `files` pattern.
+
+#### Applying environments
+
+The `ESLintRCCompat#env()` method allows users to specify an `env` settings as they would in an `.eslintrc`-style config and have globals automatically added. For example:
+
+```js
+const { ESLintRCCompat } = require("@eslint/eslintrc");
+
+const eslintrc = new ESLintRCCompat(__dirname);
+
+module.exports = [
+    "eslint:recommended",
+  
+    // load node environment
+    eslintrc.env({
+        node: true
+    })
+];
+```
+
+### Configuration Location Resolution
+
+When ESLint is executed, the following steps are taken to find the `eslint.config.js` file to use:
+
+1. If the `-c` flag is used then the specified configuration file is used. There is no further search performed.
+1. Otherwise:
+    a. Look for `eslint.config.js` in the current working directory. If found, stop searching and use that file.
+    b. If not found, search up the directory hierarchy looking for `eslint.config.js`.
+    c. If a `eslint.config.js` file is found at any point, stop searching and use that file.
+    d. If `/` is reached without finding `eslint.config.js`, then stop searching and output a "no configuration found" error.
```

Comment:
Formatting nit: Right now GitHub's Markdown renderer is merging these lines together into one paragraph. If you want to preserve the `<ol>`, this I believe it will translate this into i-iv sub-numbering because of the indentation as it did in the File Pattern Resolution section below:

```suggestion
    1. Look for `eslint.config.js` in the current working directory. If found, stop searching and use that file.
    1. If not found, search up the directory hierarchy looking for `eslint.config.js`.
    1. If a `eslint.config.js` file is found at any point, stop searching and use that file.
    1. If `/` is reached without finding `eslint.config.js`, then stop searching and output a "no configuration found" error.
```